src/sched/lthread.c

```diff
@@ -233,13 +231,31 @@ void lthread_run(void)
             dequeued = 0;
             if (mpmc_dequeue(&__scheduler_queue, (void**)&lt))
             {
+                SGXLKL_ASSERT(!(lt->attr.state & BIT(LT_ST_EXITED)));
+
                 dequeued++;
                 pauses = sleepspins;
-                a_dec(&schedqueuelen);
                 SGXLKL_TRACE_THREAD(
                     "[%4d] lthread_run(): lthread_resume (dequeue)\n",
                     lt ? lt->tid : -1);
                 _lthread_resume(lt);
+
+                // Bail out if there is a terminating scheduler, and we are not
+                // it.
+                if (_lthread_terminating_scheduler &&
+                    _lthread_terminating_scheduler != sched)
```

Comment:
Two reads of an `_Atomic` make me nervous.  I *think* this is okay, because `_lthread_terminating_scheduler` can't be modified in such a way that the second condition becomes true, but I wouldn't need to think that through if we did a single read.

Reply:
What code do you have in mind?

Reply:
Read the atomic variable to a local and use that.

Reply:
Done.